gandalf.js

```diff
@@ -1,49 +1,47 @@
 /**
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.